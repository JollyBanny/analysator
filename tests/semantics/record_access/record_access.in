var rec : record
    x: integer;
    end;

begin
    write(rec.x);
end.