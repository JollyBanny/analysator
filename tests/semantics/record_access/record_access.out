program
└──── program block
      ├──── VAR
      │     └──── var declaration
      │           ├──── rec
      │           └──── RECORD
      │                 └──── :
      │                       ├──── x
      │                       └──── integer
      └──── Compound statement
            └──── Call statement
                  └──── write
                        └──── record access
                              ├──── rec
                              └──── x

┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│  builtins                                                                                                            │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  integer                                                                                         │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  double                                                                                          │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  char                                                                                            │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  string                                                                                          │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  boolean                                                                                         │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│  global                                                                                                              │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Var              │  rec                        │  { x:integer }                                                     │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
